### PR TITLE
feat: add loading bar to each XHR.

### PR DESCRIPTION
**Pull request by ceoaliongroo**
#27 

screen shot 2015-01-13 at 11 30 29 am

### CODE REVIEW

**Comment by ceoaliongroo** [comment]
@amitaibu, could you review it?

thanks

File: .travis.yml [hunk 1]
```diff
@@ -74,6 +74,8 @@ before_script:
 
 script:
   # Run PhantomCSS tests.
+  - sleep 5 # give time to grunt serve finish to load
+  - cat ~/grunt.log
```
Comment (amitaibu) [inline .travis.yml]: remove

File: .travis.yml [hunk 2]
```diff
@@ -88,5 +90,5 @@ script:
   - ./bin/behat --tags=~@wip
 
   # Print logs.
-  - cat ~/grunt.log
+  # - cat ~/grunt.log
```
Comment (amitaibu) [inline .travis.yml]: keep this.

File: app/templates/client/app/scripts/directives/loadingbartext.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        $scope.$on('cfpLoadingBar:started', function() {
+          setLoading(true);
+        });
+
+        $scope.$on('cfpLoadingBar:completed', function() {
+          setLoading(false);
+        });
+      },
+      // Isolate scope.
+      scope: {}
```
Comment (amitaibu) [inline app/templates/client/app/scripts/directives/loadingbartext.js]: this part isn't clear

File: app/templates/client/app/scripts/directives/loadingbartext.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ * @name clientApp.directive:loadingBarText
+ * @description
+ * # loadingBarText
+ */
+angular.module('clientApp')
+  .directive('loadingBarText', function () {
+    return {
+      retrict: 'EA',
+      template: '<div class=\"splash-screen\" ng-show="isLoading">{{text}}</div>',
+      controller: function($scope) {
```
Comment (amitaibu) [inline app/templates/client/app/scripts/directives/loadingbartext.js]: what's the advantage of using a directive's controller here?

File: app/templates/client/app/scripts/directives/loadingbartext.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        $scope.text = 'loading...';
+        $scope.isLoading = false;
+
+        /**
+         * Determine if the text is showing.
```
Comment (amitaibu) [inline app/templates/client/app/scripts/directives/loadingbartext.js]: ``Indicate a loading is in process.``

File: app/templates/client/app/scripts/directives/loadingbartext.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+         *
+         * @param isLoading - Boolean
```
Comment (amitaibu) [inline app/templates/client/app/scripts/directives/loadingbartext.js]: ``@param bool isLoading``

File: app/templates/client/app/scripts/directives/loadingbartext.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+         *  True show the text, false hide it.
```
Comment (amitaibu) [inline app/templates/client/app/scripts/directives/loadingbartext.js]: ``True to show the loader.``

File: app/templates/client/app/scripts/directives/loadingbartext.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+         */
+        function setLoading(isLoading) {
+          $scope.isLoading = isLoading;
+        }
+
+        // Events to set the message when start the XHR request until is completed.
+        $scope.$on('cfpLoadingBar:started', function() {
+          setLoading(true);
```
Comment (amitaibu) [inline app/templates/client/app/scripts/directives/loadingbartext.js]: ``Set loading progress bar upon HTTP request.``

File: app/templates/client/app/scripts/directives/loadingbartext.js (hunk 6 above)
Comment (ceoaliongroo) [inline app/templates/client/app/scripts/directives/loadingbartext.js]: is need it to add first the name of the variable.
``@param isLoading bool``

File: app/templates/client/app/scripts/directives/loadingbartext.js (hunk 4 above)
Comment (ceoaliongroo) [inline app/templates/client/app/scripts/directives/loadingbartext.js]: - Isolate the construction and ui implementation into the directive. (Architecture based on Web components)
- We can easy extend the directive. 
- Inject the controller in the directive pattern will be a standard in Angular 2.0, already used in AngularDart

File: .travis.yml [hunk 9]
```diff
@@ -74,6 +74,7 @@ before_script:
 
 script:
   # Run PhantomCSS tests.
+  - sleep 5 # give time to grunt serve finish to load
```
Comment (amitaibu) [inline .travis.yml]: can you try and remove this - I'd like to see the travis that fails.

**Comment by amitaibu** [comment]
Merged, thanks.

**Pull request closed without merging**